api/src/database/helpers/number/types.ts

```diff
@@ -0,0 +1,24 @@
+import type { Knex } from 'knex';
+import { DatabaseHelper } from '../types.js';
+
+// move to @directus/types
+const NUMERIC_TYPES = ['bigInteger', 'decimal', 'float', 'integer'] as const;
+
+export type NumericValue = number | bigint;
+export type NumericType = (typeof NUMERIC_TYPES)[number];
+export type NumberInfo = {
+	type: NumericType;
+	precision: number | null;
+	scale: number | null;
+};
+
+export abstract class NumberDatabaseHelper extends DatabaseHelper {
+	// maybe call this "addSearchCondition"?
+	orWhere(dbQuery: Knex.QueryBuilder, collection: string, name: string, value: NumericValue): Knex.QueryBuilder {
```

Comment:
Since `orWhere` is pretty generic what do you think of something like `addSearchCondition`